tests/sentry/workflow_engine/handlers/condition/test_slow_condition_query_handlers.py

```diff
@@ -40,13 +40,183 @@ def test_batch_query(self):
+            group_ids={self.event3.group_id},
+            start=self.start,
+            end=self.end,
+            environment_id=self.environment2.id,
+            filters=[{"key": "biz", "match": "nc", "value": "b"}],
+        )
+        assert batch_query == {self.event3.group_id: 0}
+
+    def test_batch_query__tag_conditions__is_set(self):
+        batch_query = self.handler().batch_query(
+            group_ids={self.event.group_id, self.event2.group_id, self.perf_event.group_id},
+            start=self.start,
+            end=self.end,
+            environment_id=self.environment.id,
+            filters=[{"key": "foo", "match": "is"}],
+        )
+        assert batch_query == {
+            self.event.group_id: 1,
+            self.event2.group_id: 1,
+            self.perf_event.group_id: 1,
+        }
+
+    def test_batch_query__tag_conditions__is_not_set(self):
+        batch_query = self.handler().batch_query(
+            group_ids={self.event.group_id, self.event2.group_id, self.perf_event.group_id},
+            start=self.start,
+            end=self.end,
+            environment_id=self.environment.id,
+            filters=[{"key": "foo", "match": "ns"}],
+        )
+        assert batch_query == {
+            self.event.group_id: 0,
+            self.event2.group_id: 0,
+            self.perf_event.group_id: 0,
+        }
+
+    def test_batch_query__tag_conditions__is_in(self):
+        batch_query = self.handler().batch_query(
+            group_ids={self.event.group_id, self.event2.group_id, self.perf_event.group_id},
+            start=self.start,
+            end=self.end,
+            environment_id=self.environment.id,
+            filters=[{"key": "region", "match": "in", "value": "US,EU"}],
+        )
+        assert batch_query == {
+            self.event.group_id: 1,
+            self.event2.group_id: 1,
+            self.perf_event.group_id: 0,
+        }
+
+    def test_batch_query__tag_conditions__is_not_in(self):
+        batch_query = self.handler().batch_query(
+            group_ids={self.event.group_id, self.event2.group_id, self.perf_event.group_id},
+            start=self.start,
+            end=self.end,
+            environment_id=self.environment.id,
+            filters=[{"key": "region", "match": "nin", "value": "US,EU"}],
+        )
+        assert batch_query == {
+            self.event.group_id: 0,
+            self.event2.group_id: 0,
+            self.perf_event.group_id: 1,
+        }
+
+    def test_batch_query__tag_conditions__invalid(self):
+        with pytest.raises(ValueError):
+            self.handler().batch_query(
+                group_ids={self.event.group_id, self.event2.group_id, self.perf_event.group_id},
+                start=self.start,
+                end=self.end,
+                environment_id=self.environment.id,
+                filters=[{"key": "region", "match": "asdf", "value": "U"}],
+            )
+
+    def test_batch_query__attribute_conditions(self):
```

Comment:
nice tests cases, i really like how these are structured and have focused assertions per case 🎉